tests/UI/campaigns/regression/currencies/computingPrecision_FO.js

```diff
@@ -0,0 +1,535 @@
+
+// Import pages
+
+// BO pages
+const dashboardPage = require('@pages/BO/dashboard');
+const cartRulesPage = require('@pages/BO/catalog/discounts');
+const addCartRulePage = require('@pages/BO/catalog/discounts/add');
+const localizationPage = require('@pages/BO/international/localization');
+const currenciesPage = require('@pages/BO/international/currencies');
+const addCurrencyPage = require('@pages/BO/international/currencies/add');
+const ordersPage = require('@pages/BO/orders');
+const sqlManagerPage = require('@pages/BO/advancedParameters/database/sqlManager');
+const addSqlQueryPage = require('@pages/BO/advancedParameters/database/sqlManager/add');
+const viewSqlQueryPage = require('@pages/BO/advancedParameters/database/sqlManager/view');
+
+// FO pages
+const homePage = require('@pages/FO/home');
+const searchResultsPage = require('@pages/FO/searchResults');
+const foLoginPage = require('@pages/FO/login');
+const productPage = require('@pages/FO/product');
+const cartPage = require('@pages/FO/cart');
+const checkoutPage = require('@pages/FO/checkout');
+const orderConfirmationPage = require('@pages/FO/checkout/orderConfirmation');
+
+
+// Import expect from chai
+const {expect} = require('chai');
+
+// Import demo data
+const {PaymentMethods} = require('@data/demo/paymentMethods');
+const {DefaultAccount} = require('@data/demo/customer');
+const {Currencies} = require('@data/demo/currencies');
+
+const {Products} = require('@data/demo/products');
+
+
+// Import faker data
+const CartRuleFaker = require('@data/faker/cartRule');
+
+const percentCartRule = new CartRuleFaker(
+  {
+    name: 'discount15',
+    code: 'discount15',
+    discountType: 'Percent',
+    discountPercent: 15,
+  },
+);
+
+const giftCartRule = new CartRuleFaker(
+  {
+    name: 'freeGiftMug',
+    code: 'freeMug',
+    discountType: 'None',
+    freeGift: true,
+    freeGiftProduct: Products.demo_13,
+  },
+);
+
+// Create sql query data to get last order discount and total price
+const sqlQueryData = {
+  name: 'Discount and TTC from last order',
+  sqlQuery: 'SELECT total_discounts, total_paid_tax_incl '
+    + 'from  ps_orders '
+    + 'ORDER BY id_order '
+    + 'DESC LIMIT 1;',
+};
+
+// Init data for the order
+const orderToMake = {
+  product: Products.demo_3,
+  quantityToOrder: 4,
+  percentDiscountValue: 20.678,
+  giftDiscountValue: giftCartRule.freeGiftProduct.price,
+  ttcPrice: 117.178,
+};
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'regression_currencies_computingPrecision_FO';
+
+// Browser and tab
+let browserContext;
+let page;
+
+/*
+Change euro decimal to 3
+Create 2 cart rules
+  1. 15% discount
+  2. Product 'Mug today is a good day' as a gift product
+Place an order in FO with the cart rules created
+  1. Check discount value after first cart rule added
+  2. Check discount value after second cart rule added
+  3. CheckTTC after second cart rule added
+  4.Finish the order
```

Comment:
```suggestion
  4. Finish the order
```
For the sake of my OCD :grin:

Reply:
done thanks